src/stx/common/trex_rx_timesync.cpp

```diff
@@ -8,89 +8,92 @@
 
 void RXTimesync::handle_pkt(const rte_mbuf_t *m, int port) {
     if (m_timesync_engine->getTimesyncMethod() == TimesyncMethod::PTP) {
-        uint16_t rx_tstamp_idx = 0;
 
-        if (hardware_timestamping_enabled) {
-            rx_tstamp_idx = m->timesync;
-        }
+        timespec t { 0, 0 };
 
         if (m->pkt_len < ETH_HDR_LEN)
             return;
 
-        uint64_t m_timestamp = 0;
-        if (CGlobalInfo::m_options.is_timesync_rx_callback_enabled()) {
-            m_timestamp = m->timestamp;
+        if (hardware_timestamping_enabled && (m->ol_flags & PKT_RX_IEEE1588_TMST)) {
```

Comment:
how do we know those incoming packets will have this flag?

Reply:
If card has hardware timestamping implemented in dpdk, this flag will be set if packet is flaged as timestamped.
http://code.dpdk.org/dpdk/v18.11/source/drivers/net/ixgbe/ixgbe_rxtx.c#L1433